contractcourt/channel_arbitrator.go

```diff
@@ -325,6 +324,10 @@ type ChannelArbitrator struct {
 	// to do its duty.
 	cfg ChannelArbitratorConfig
 
+	// blocks is a queue that the chain arbitrator will use to deliver
+	// blocks to the chanArb.
+	blocks *queue.ConcurrentQueue
```

Comment:
Why not just a normal unbuffered (or buffer of 1) channel? Would serve to reduce the memory footprint over all.

Reply:
Started off with this then reverted to a queue to see whether it fixed the itest hangs. Since it seems like it's the lock, will go back to a single buffered channel.